dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AnalyticsController.java

```diff
@@ -427,18 +433,23 @@ public void getRawDataCsv(
         @RequestParam( required = false ) boolean skipData,
         @RequestParam( required = false ) boolean hierarchyMeta,
         @RequestParam( required = false ) boolean showHierarchy,
+        @RequestParam( required = false ) boolean includeOrgUnitNames,
         @RequestParam( required = false ) DisplayProperty displayProperty,
         @RequestParam( required = false ) IdScheme outputIdScheme,
         @RequestParam( required = false ) IdScheme inputIdScheme,
         @RequestParam( required = false ) String userOrgUnit,
         DhisApiVersion apiVersion,
         Model model,
-        HttpServletResponse response ) throws Exception
+        HttpServletResponse response )
+        throws Exception
     {
-        DataQueryParams params = dataQueryService.getFromUrl( dimension, null, null, null, null, startDate, endDate, skipMeta, skipData, false, false, hierarchyMeta,
-            false, false, false, showHierarchy, false, displayProperty, outputIdScheme, inputIdScheme, false, null, null, userOrgUnit, true, apiVersion );
+        DataQueryParams param = dataQueryService.getFromUrl( dimension, null, null, null, null, startDate, endDate, skipMeta, skipData, false, false, hierarchyMeta,
+                false, false, false, showHierarchy, false, displayProperty, outputIdScheme, inputIdScheme, false, null, null, userOrgUnit, true, apiVersion );
+
+        DataQueryParams params = DataQueryParams.newBuilder( param ).withIncludeOrgUnitNames( includeOrgUnitNames ).build();
 
```

Comment:
Change this, instead include include org unit names in the .getFromUrl method signature directly.